notebooks/writing_a_trace_transform_cpu_offloading.ipynb

```diff
@@ -0,0 +1,752 @@
+{
```

Comment:
<p>[nitpicking] Is <a href="https://github.com/Lightning-AI/lightning-thunder/blob/e88966f2f35433e167f40d26e1330684b8ee3fc1/thunder/core/transform_common.py#L337" target="_blank">thunder.core.transform_common.Transform</a> explicitly exposed  in <code>thunder</code> namespace? </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Lightning-AI/lightning-thunder/pull/966/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='2'/>

Reply:
It isn't, but surprisintly even `jit` isn't 😮 

https://github.com/Lightning-AI/lightning-thunder/blob/e88966f2f35433e167f40d26e1330684b8ee3fc1/thunder/__init__.py#L95-L128

Reply:
This comment is orthogonal to this PR itself, but IMHO it would make sense to avoid defining methods and/or classes in dunder init files and it would be better to such files just have imports and dunder all.
This is because I'm speculating that this exposure of `thunder.core.transform_common.Transform` in `thunder` namespace is not quite expected.

Reply:
My understanding of dunder all is rather to avoid name collisions that start import could cause, while using it as a list of public features is understandable

Reply:
As per pep-8 style guide https://peps.python.org/pep-0008/#public-and-internal-interfaces,

> To better support introspection, modules should explicitly declare the names in their public API using the __all__ attribute. Setting __all__ to an empty list indicates that the module has no public API.

Reply:
yup (how much would we obey pep-8 style guide?)

Reply:
There is no harm in improving.